lib/whitehall_importer/migrate_assets.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class MigrateAssets
+    attr_reader :whitehall_import
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(whitehall_import)
+      @whitehall_import = whitehall_import
+    end
+
+    def call
+      whitehall_import.assets.reject { |asset| asset.state == "processed" }.each do |whitehall_asset|
```

Comment:
Might be better to have this as a scope rather than a reject method? also I'd imagine we only want things that are not_processed as (unlikely as it could be) wouldn't want something that is processing.

Reply:
Having difficulty getting this working with associations - see the "WIP: add a 'pending' scope" commit. Would appreciate any pointers! Agree the code will look much nicer when this is done 👍

Reply:
For the record, you provided some ideas here: https://github.com/alphagov/content-publisher/commit/0a2f64502c1a7e277fe0efad744176d8c0b2b989

> All but the method with the select will fail the tests as the scope and
association based approaches will make a fresh db query and that will
mean any mocks that are set won't have an affect.
> It also needs to use a create approach for the tests as with build any
methods that require fresh queries (scope or different has_many) won't
return any results.
> We already get pre-defined scopes for each of the enum values, which is
how I was able to do a scope of just `pending.or(migration_failed)`

I've ended up abandoning the scopes idea and have opted for a `migratable_assets` method on the `DocumentImport` class - a variation on what you have provided in your spike - which returns a subset of its `assets`.